lib/msf/core/db_manager/vuln.rb

```diff
@@ -150,6 +150,7 @@ def report_vuln(opts)
         case opts[:proto].to_s.downcase # Catch incorrect usages, as in report_note
         when 'tcp','udp'
           proto = opts[:proto]
+          sname = opts[:sname]
```

Comment:
From a quick look; it seems like `sname` isnt always present to `report_vuln` calls - likewise the documentation suggests only these keys are required:

```
  #
  # opts MUST contain
  # +:host+:: the host where this vulnerability resides
  # +:name+:: the friendly name for this vulnerability (title)
  #
  # opts can contain
  # +:info+::   a human readable description of the vuln, free-form text
  # +:refs+::   an array of Ref objects or string names of references
  # +:details:: a hash with :key pointed to a find criteria hash and the rest containing VulnDetail fields
  #
```

Does this code need to be changed to accommodate for the scenario of `sname` not being present? And potentially an update to the `opts can contain` section

Reply:
When `sname` is not present, we're going to fall back to the same situation we had before.
Right now the fix is based upon what the Redis login scanner passes down.

My gut feeling says to update the documentation and ensure `:sname` is set.
WDYT?

Reply:
I only briefly looked at the code so I might be wrong; when `sname` isn't present - won't it now try doing a lookup of `name: nil`? Whilst before it wouldn't have that filter applied:

```diff
-        service = host.services.where(port: opts[:port].to_i, proto: proto).first_or_create
+        service = host.services.where(name: sname, port: opts[:port].to_i, proto: proto).first_or_create
```

So we can in theory get very different results now, i.e. if there was a service registered on that port but we just don't know the name

Reply:
Tests failed with lookup when `sname` is not present.
So I have adjusted the code to me more clever in handling this parameter. If it's provided by the caller, we will use it and scope down the search for the service.

If it's omitted, we retain the original behavior which should cover both cases.